dotCMS/src/main/java/com/dotcms/publisher/util/DependencySet.java

```diff
@@ -108,7 +116,43 @@ private boolean addOrClean ( String assetId, Date assetModDate, Boolean cleanFor
             for (Environment env : envs) {
 				PushedAsset asset;
 				try {
-					asset = APILocator.getPushedAssetsAPI().getLastPushForAsset(assetId, env.getId());
+					if(endpointIds == null){
+						//this code allows to find if the asset should be included or not in the dependencies depending of the assetId, environment Id and endpoint Ids 
+						List<PublishingEndPoint> allEndpoints = APILocator.getPublisherEndPointAPI().findSendingEndPointsByEnvironment(env.getId());
+						List<String> endpoints = new ArrayList<String>();
+		                //Filter Endpoints list
+		                for(PublishingEndPoint ep : allEndpoints) {
+		                    if(isStatic && ep.isEnabled() && AWSS3Publisher.PROTOCOL_AWS_S3.equals(ep.getProtocol())) {
+		                    	//If the isStatic variable is true then get all the static endpoints
+		                    	endpoints.add(ep.getId());
+		                        //Set that class name of the puh publisher used by these endpoints
+		                        if(publisher == null){
+		                        	publisher=AWSS3Publisher.class.getName();
+		                        }
+		                    }else if(!isStatic && ep.isEnabled() && !AWSS3Publisher.PROTOCOL_AWS_S3.equals(ep.getProtocol())) {
+		                    	//If the isStatic variable is false then get all the no static endpoints
+		                    	endpoints.add(ep.getId());
+		                    	//Set that class name of the puh publisher used by these endpoints
+		                    	if(publisher == null){
+		                        	publisher=PushPublisher.class.getName();
+		                        }
+		                    }
+		                }
+		                //comma separated string with the list of endpoint ids 
+		                endpointIds = StringUtils.join(endpoints,",");
+		                if(!env.getPushToAll()) {
+		                    if(endpointIds != null && endpointIds.indexOf(",") != -1){
+		                        endpointIds = endpointIds.substring(0, endpointIds.indexOf(","));
+		                    }
+		                }
+					}
+					if(endpointIds != null){
```

Comment:
If the "endpointIds" variable is null, we can just have one single statement. We don't need to IF clause.